rust/src/snmp/detect.rs

```diff
@@ -17,42 +17,263 @@
 
 // written by Pierre Chifflier  <chifflier@wzdftpd.net>
 
-use crate::snmp::snmp::SNMPTransaction;
+use super::snmp::{SNMPTransaction, ALPROTO_SNMP};
+use crate::detect::uint::{
+    rs_detect_u32_free, rs_detect_u32_match, rs_detect_u32_parse, DetectUintData,
+};
+use crate::detect::{
+    DetectBufferSetActiveList, DetectHelperBufferMpmRegister, DetectHelperBufferRegister,
+    DetectHelperGetData, DetectHelperKeywordRegister, DetectSignatureSetAppProto, SCSigTableElmt,
+    SigMatchAppendSMToList, SIGMATCH_INFO_STICKY_BUFFER, SIGMATCH_NOOPT,
+};
+use std::ffi::{c_int, c_void};
+
+static mut G_SNMP_VERSION_KW_ID: c_int = 0;
+static mut G_SNMP_VERSION_BUFFER_ID: c_int = 0;
+static mut G_SNMP_PDUTYPE_KW_ID: c_int = 0;
+static mut G_SNMP_PDUTYPE_BUFFER_ID: c_int = 0;
+static mut G_SNMP_USM_BUFFER_ID: c_int = 0;
+static mut G_SNMP_COMMUNITY_BUFFER_ID: c_int = 0;
+
+#[no_mangle]
+unsafe extern "C" fn snmp_detect_version_setup(
```

Comment:
naming: no_mangle and extern C should follow C naming conventions.

Reply:
As this isn't `pub`, and just a function pointer, you could drop the `no_mangle` and use Rust conventions. Meaning you could drop the prefixes in the function name as well, as the visibility is just this module.

Reply:
What does the `no_mangle` do ?
I thought I needed it for the ABI

Reply:
No mangle just keeps the name as-is in the symbol table so linkers can find it.  If you just point to it using a function pointer its not needed and best let Rust do its name mangling.

The `extern` is for the ABI.

Reply:
Thanks Jason, makes sense